lib/Cake/View/Helper/FormHelper.php

```diff
@@ -1389,7 +1389,7 @@ protected function _divOptions($options) {
 		} elseif (is_array($div)) {
 			$divOptions = array_merge($divOptions, $div);
 		}
-		if ($this->_extractOption('required', $options) !== false &&
+		if ($this->_extractOption('required', $options) !== false ||
```

Comment:
From a quick grasp, that makes sense. Also the documetnation states this

> Since 2.3 the HTML5 required attribute will also be added to the input based on validation rules. __You can explicitly set required key in options array to override it for a field.__

https://book.cakephp.org/2.0/en/core-libraries/helpers/form.html#FormHelper::input